### PR TITLE
Getting pycbc_multi_inspiral to run to completion

### PR DESCRIPTION
**Pull request by a-r-williamson**
This commit fixed some code incompatibilities and allowed `pycbc_multi_inspiral` to run to completion (in its state a few months back). Additional changes may need to be made after recent pycbc development, so consider this a jumping off point.

DO NOT MERGE!

### CODE REVIEW

**Comment by spxiwh** [comment]
:thumbsup:

**Comment by a-r-williamson** [comment]
@cmbiwer @spxiwh 

I have made small changes to `pycbc_generate_hwinj` and `pycbc_inference` in response to the Travis build check. In them the option group `insert_strain_option_group_multi_ifo` was imported, which I had previously edited to include a `--frame-type` option. This conflicted with `--frame-type` options in `pycbc_generate_hwinj` and `pycbc_inference`. By removing them from these files, I believe this will not change their behaviour, as the option is now loaded along with the rest of `insert_strain_option_group_multi_ifo`. Can you confirm this change is ok?

**Comment by spxiwh** [comment]
I'm happy with these changes and am ready to merge this. You still have the "work in progress" label applied though. Should this be merged now, or are you waiting for something?

**Comment by a-r-williamson** [comment]
@spxiwh Let me run another test first. I'll remove the label and ask for a merge once this is done.

Thanks.

**Comment by a-r-williamson** [comment]
@spxiwh I am happy for this to be merged if you are, thanks.

**Comment by spxiwh** [comment]
Hi @a-r-williamson, thanks for this. I only have one confusion with the code itself, and then there's the complaints raised by the various code checkers: You add a new write_to_hdf function (yay, for getting rid of XML) for the MultiEventManager class. However, this seems to reflect rather too closely the SingleEventManager version and is neglecting multi-detector information. Currently this code is not doing anything coherent, so it should basically output the same format as the SingleEventManager, but for multiple-ifos. But to do that, the output function needs to know about the ifo. So H1 triggers should be written to hdf_fp['H1/snr'] and similar and L1 triggers to hdf_fp['L1/SNR'] ... currently you write everything to hdf_fp['H1L1/SNR'] and it is then not clear to me which triggers are L1 and which are H1 .... We will need a different output function (and "finalize" function) when this gets made coherent, which would then write to ['H1L1/coherent_snr'], but that will need some larger-scale changes.

Landscape, codacy and QuantifiedCode are all suggesting changes here. Can you look over these and try to address them.

The landscape "too-many-[arguments,variables,statements]" can be ignored, but the unused variables, or undeclared exception issues should be fixed (if this is copied from crappy older code elsewhere, that predates these fixes, please fix the issue in the original code as well ... This keeps with our policy of slowly improving the code readability and quality!). The "code is too complex" warning should not really be ignored, but if there is no obvious way to split this function up further, then also ignore it.

File: pycbc/events/events.py [hunk 1]
```diff
@@ -586,15 +586,149 @@ def finalize_template_events(self, perform_coincidence=True,
...
+                                      shuffle=True)
 
+        self.events.sort(order='template_id')
+        th = numpy.array([p['tmplt'].template_hash for p in \
+                          self.template_params])
+        tid = self.events['template_id']
+        f = fw(outname, ''.join(self.opt.instruments))
+
+        if len(self.events):
+            ifo_str = self.opt.instruments[0]
```
Comment (spxiwh) [inline pycbc/events/events.py]: The active instruments are already a property of the MultiEvent class

**Comment by a-r-williamson** [comment]
@spxiwh I've tried to address these issues, let me know if there's still a problem, thanks.

**Pull request merged**